diesel/src/hlist.rs

```diff
@@ -0,0 +1,88 @@
+use std::fmt::{Debug, Formatter, Error as FmtError};
+
```

Comment:
We should probably add some (module-level) docs here. At least for contributors looking to understand diesel's inner workings. [Like this](https://github.com/lloydmeta/frunk/blob/4d688a460115474d915be7a7d351b508c478ab96/core/src/hlist.rs) maybe.